src/view/DashboardView.java

```diff
@@ -0,0 +1,239 @@
+package view;
+
+import interface_adapter.dashboard.DashboardState;
+import interface_adapter.dashboard.DashboardViewModel;
+import interface_adapter.dashboard.ProjectData;
+import interface_adapter.add_project.AddProjectController;
+import interface_adapter.send_notification.NotificationController;
+
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.text.ParseException;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import javax.swing.text.MaskFormatter;
+
+public class DashboardView extends JFrame implements PropertyChangeListener {
+    private final DashboardViewModel dashboardViewModel;
+    private JPanel dashboardPanel;
+    private ArrayList<ProjectData> projectsList;
+    private final ScheduledExecutorService schedule = Executors.newScheduledThreadPool(1);
+    private ScheduledFuture<?> scheduledFuture = null;
+
+    public DashboardView(DashboardViewModel dashboardViewModel, AddProjectController addProjectController,
+                         NotificationController notificationController) {
+        this.dashboardViewModel = dashboardViewModel;
+
+        DashboardState dashboardState = dashboardViewModel.getState();
+
+        dashboardViewModel.addPropertyChangeListener(this);
+
+        projectsList = dashboardState.getProjects();
+
+        displayAllProjects();
+
+        setTitle("Project Dashboard");
+        setSize(600, 400);
+        setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        setLayout(new BorderLayout());
+        setResizable(false);
+
+        dashboardPanel = new JPanel();
+        dashboardPanel.setLayout(new BoxLayout(dashboardPanel, BoxLayout.Y_AXIS));
+
+        JScrollPane scrollPane = new JScrollPane(dashboardPanel);
+        scrollPane.setVerticalScrollBarPolicy(ScrollPaneConstants.VERTICAL_SCROLLBAR_ALWAYS);
+
+        add(scrollPane, BorderLayout.CENTER);
+
+        updateEmptyDashboardLabel();
+
+        JButton addProjectButton = new JButton("Add Project");
+        JButton toggleNotifications = new JButton("Notifications Off");
```

Comment:
I think because you copied over William's Dashboard view, a comment that indicates seperation from his code to yours would be convinient for others to review your code